### PR TITLE
Update and submission: contacts/contacts-cli

### PR DESCRIPTION
**Pull request by herrbischoff**
#### Description

Since `contacts` is unmaintained for years now, update the port for it to the last version of a fork that used to be maintained. Sorry for the commit ID, there is no proper release with the needed UTF-8 fixes. Also, add `contacts-cli` as a modern alternative. Since both install a `contacts` binary, mark them as conflicting.

Two commits as those are two submissions. If needed, I can squash them into one.

Existing Trac ticket: https://trac.macports.org/ticket/58472

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.5 18F132
Xcode 10.2.1 10E1001

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by herrbischoff** [comment]
I don't understand why the tests fail. There is no error in the logs, yet the bash script exits with `1`. Could someone with more knowledge about the build setup take a look please?

**Comment by pmetzger** [comment]
Just curious, but why have both, and not just one of them (presumably the maintained one)? Especially if they have the same functionality?

**Comment by pmetzger** [comment]
I'm re-running the CI btw. Lets see what the error is.

File: sysutils/contacts/Portfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        dhess contacts a4db4d86caa37553b8c4c6f70bd9d91bd34aa705
```
Comment (mf2k) [inline sysutils/contacts/Portfile]: This makes the version number `a4db4d86caa37553b8c4c6f70bd9d91bd34aa705` which is not what we want. Please use a release version or manually set the version to something sensible.

**Comment by herrbischoff** [comment]
> Just curious, but why have both, and not just one of them (presumably the maintained one)? Especially if they have the same functionality?

They *almost* have the same functionality. `contacts-cli` is missing some advanced formatting options that `contacts` does have. Some users will probably want to use the old version. In the end, it's about not throwing out something that works for the time being. Also, this mirrors what Homebrew is doing.

**Comment by cjones051073** [comment]
@herrbischoff ping. Please can you make the requested changes ?

**Comment by herrbischoff** [comment]
Thanks for the ping. Completely forgot about this. I will get on it as soon as I can.

File: sysutils/contacts/Portfile (hunk 1 above)
Comment (herrbischoff) [reply sysutils/contacts/Portfile]: The issue is, there is no release version containing this. As this project is also no longer maintained, I would have to create a pseudo-release myself, which I don't want to do. What to do? Just manually set the version to the last release version? Invent a version number? What is the best practice here?

Comment (herrbischoff) [reply sysutils/contacts/Portfile]: Perfect, thanks!

**Comment by cjones051073** [comment]
> The CI error is:
> 
> ```
> install contacts /opt/local/bin
> install: /opt/local/bin/contacts: Operation not permitted
> ```
> 
> I'm not sure how to fix that though.

I cannot access the logs it seems to check, but is it perhaps that DESTROOT is being ignore, and the install step is trying to write directly to ${prefix} when it should not be ?

**Comment by herrbischoff** [comment]
Thanks for the pointers. I will look into it over the weekend. I’m super busy right now. Just wanted to let you know the status.

**Comment by neverpanic** [comment]
@herrbischoff Any progress? I guess this would be nice to have and I'd really like to merge it – it seems there's not much work left other than making sure that it installs files in `${destroot}`.

**Comment by neverpanic** [comment]
Actually, looking at it is seems you can get away with setting `PREFIX=${destroot}${prefix}`.

**Comment by pmetzger** [comment]
Does someone want to adopt this?

**Comment by pmetzger** [comment]
I am presuming no one wishes to adopt this. I am therefore going to close it.

**Pull request closed without merging**